core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -588,6 +612,24 @@ private SqlNode getIdentifier(SqlBasicCall intervalOperand) {
     return ((SqlBasicCall) aggCall).operand(0);
   }
 
+  private void formatDate(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    final SqlWriter.Frame formatDate = writer.startFunCall("TO_VARCHAR");
+    call.operand(1).unparse(writer, leftPrec, rightPrec);
+    writer.print(",");
+    switch (((SqlLiteral) call.operand(0)).getValueAs(String.class)) {
+    case "EEEE":
+    case "E4":
+    case "EEE":
```

Comment:
Please use DATE_TIME_FORMAT_MAP to convert from EEEE to DY

Reply:
done